test/NuGet.Core.Tests/NuGet.Frameworks.Test/CompatibilityTests.cs

```diff
@@ -42,15 +42,36 @@ public class CompatibilityTests
         [InlineData("net5.1", "netcoreapp3.1", true)]
         [InlineData("net6.0", "netcoreapp3.1", true)]
 
-        // net5.0-<osname> is not supported yet
-        [InlineData("net5.0-android", "net5.0", false)]
-        [InlineData("net5.0-android", "netcoreapp3.1", false)]
-        [InlineData("net5.0-ios", "net5.0", false)]
-        [InlineData("net5.0-macos", "net5.0", false)]
-        [InlineData("net5.0-tvos", "net5.0", false)]
-        [InlineData("net5.0-watchos", "net5.0", false)]
-        [InlineData("net5.0-windows", "net5.0", false)]
+        // net5.0-<osname> is compatible with net5.0 but not the inverse
+        [InlineData("net5.0-android", "net5.0", true)]
+        [InlineData("net5.0-android", "netcoreapp3.1", true)]
+        [InlineData("net5.0-ios", "net5.0", true)]
+        [InlineData("net5.0-macos", "net5.0", true)]
+        [InlineData("net5.0-tvos", "net5.0", true)]
+        [InlineData("net5.0-watchos", "net5.0", true)]
+        [InlineData("net5.0-windows", "net5.0", true)]
+        [InlineData("net5.0", "net5.0-android", false)]
+        [InlineData("net5.0", "net5.0-ios", false)]
+        [InlineData("net5.0", "net5.0-macos", false)]
+        [InlineData("net5.0", "net5.0-tvos", false)]
+        [InlineData("net5.0", "net5.0-watchos", false)]
         [InlineData("net5.0", "net5.0-windows", false)]
+        [InlineData("net5.0-mac os", "net5.0", false)]
+        [InlineData("net5.0-macos-14", "net5.0", false)]
+        [InlineData("net5.0-macos-14..0", "net5.0", false)]
```

Comment:
How does this parse?
What are the platform/platform version?

Reply:
These parse to Unsupported. I'll document that.